amethyst_assets/examples/hl.rs

```diff
@@ -148,10 +149,22 @@ impl State {
 
                 Some(State::Loading(progress))
             }
-            State::Loading(progress) => if progress.is_complete() {
-                Some(State::SomethingElse)
-            } else {
-                Some(State::Loading(progress))
+            State::Loading(progress) => match progress.complete() {
+                Completion::Complete => Some(State::SomethingElse),
+                Completion::Failed => {
+                    eprintln!("Asset loading failed!");
```

Comment:
Could you use the log crate?
error!("..")

Reply:
No, there's no logging in these examples. Not necessary here.